### ISSUE TITLE
sbJSON association mapping error

### ISSUE THREAD

**Issue by jlblcc** [adiwg/mdTranslator#149]
"Produced" is actually mapping to "prodcutOf", should be mapping to "project".  Looks like "productOf" is mapping to "product" when translating from the product. This should be mapping to "parentProject".  See the project example below for reference:

SB Item: https://www.sciencebase.gov/catalog/item/54b83e7de4b03ff52703c997?community=LC+MAP+-+Landscape+Conservation+Management+and+Analysis+Portal

Need to get this fixed ASAP.

**Comment by zephyrr** [comment]
Let me broaden this and be pretty specific as it affects both directions.

In ScienceBase suppose a Project is correctly linked to a Project with a (bidirectional) Associated Item type of "produced", which looks like "product of" from the Product end.

Imported via the api to mdJSON in mdEditor, this becomes:
* Project has Associated Resource link to Product, of type "productOf" rather than "product"
* Product has Associated Resource link to Project, of type "product" rather than "parentProject"

Basically, the translation gets the direction backward, AND uses productOf (which is not an allowed type in mdEditor) rather than "parentProject" (which is allowed).

**Comment by zephyrr** [comment]
I couldn't check in the other direction, but let's be sure we get it working both directions!

**Comment by jlblcc** [comment]
I think we've got it figured out. Patched release forthcoming.

**Comment by jlblcc** [comment]
![image](https://user-images.githubusercontent.com/5833603/31418074-4687d238-ade0-11e7-9fd3-5122c152941a.png)

**Comment by stansmith907** [comment]
ADIwg tools mdJson/mdTranslator/mdEditor as well as ISO use a mono-directional relationship, i.e. all relationships are expressed 'how the associated item relates to the main resource (resource being described by the metadata)'.  ScienceBase sbJson uses bi-directional relationships to support webpage display and navigation in addition to basic metadata description.  I can get confusing trying to keep the two systems in sync. 

Josh and I revisited and adjusted the two translation tables.  Hopefully this will solve the problems.

Code for mdTranslator's sbJson reader sbJson linkType to ISO associationTypeCode translation tables:
````ruby
# mdJson/mdTranslator view point is always: how the associated resource relates to the main resource
# dropped 'copiedFrom', 'copiedInfo' from translation tables

# scienceBaseId = relatedItemId (forward)
# assoc2main = the associated resource is (a) ___ of the main resource
# example = the associated resource is 'constituentOf' the main resource
# so the adiwg equivalent is 'ifPartOf'
@association_sb2adiwg_assoc2main = [
   {sb: 'alternate', adiwg: 'alternate'},
   {sb: 'constituentOf', adiwg: 'isPartOf'},
   {sb: 'derivativeOf', adiwg: 'derivativeResource'},
   {sb: 'precededBy', adiwg: 'series'},
   {sb: 'productOf', adiwg: 'product'},
   {sb: 'related', adiwg: 'crossReference'},
   {sb: 'subprojectOf', adiwg: 'subProject'}
]

# scienceBaseId = itemId (reverse)
# main2assoc = the main resource is (a) ___ of the associated resource
# example = the main resource is 'constituentOf' the associated resource
# reverse the direction for mdJson/mdTranslator which is mono-directional
# so the adiwg equivalent is 'isComposedOf'
@association_sb2adiwg_main2assoc = [
   {sb: 'alternate', adiwg: 'alternate'},
   {sb: 'constituentOf', adiwg: 'isComposedOf'},
   {sb: 'derivativeOf', adiwg: 'source'},
   {sb: 'precededBy', adiwg: 'series'},
   {sb: 'productOf', adiwg: 'parentProject'},
   {sb: 'related', adiwg: 'crossReference'},
   {sb: 'subprojectOf', adiwg: 'parentProject'}
]
````

Code for mdTranslator's sbJson reader to determine relationship direction.
````ruby
# determine relationship direction
# forward: how the associated resource relates to the main resource
# ... in other words - the relationship is defined in terms of the associated resource
# ... example: the associated resource is a 'subProject' of the main resource
# reverse: how the main resource relates to the associated resource
# ... in other words - the relationship is defined in terms of the main resource
# ... example: the main resource is the 'parentProject' of the associated resource
# all mdJson/mdTranslator relationships must be expressed as forward
forward = nil
if hItem.has_key?('itemId')
   forward = false if sbId == hItem['itemId']
end
if hItem.has_key?('relatedItemId')
   forward = true if sbId == hItem['relatedItemId']
end
if forward.nil?
   hResponseObj[:readerExecutionMessages] << 'Main ScienceBase id was not referenced in related item'
   return hMetadata
end
````

Also note that the associationType in mdEditor is open for editing, you are not limited to the choices in the dropdown.  Also note that the sbJson linkTypes of 'copiedFrom' and 'copiedInto' were dropped from the translation tables because we could not find a way to make them work.  So these two may show up in the mdEditor at which point you can change to whatever you wish or leave them be.
